main_2024.py

```diff
@@ -255,6 +276,130 @@ def main() -> int:
                 print("geolocation confidence: " + str(detection_world.confidence))
                 print("")
 
+        # Check if all workers are alive in each manager.
+        # If a worker is dead, log the error and restart the worker.
+        # Steps: Terminate and join the worker, fill and drain the
+        # related queues, and create and start a new worker.
+        for manager in range(number_of_managers):
+            if not managers_array[manager].are_workers_alive():
+                managers_array[manager].terminate_workers()
+                managers_array[manager].join_workers()
+                managers_array[manager] = None
+
+                if manager == ManagerType.VIDEO_INPUT.value:
+                    main_logger.error(
+                        "Video Input Worker is dead, attempting to restart.",
+                        frame,
+                    )
```

Comment:
make sure to follow the docs for the logging module, we need to get a new frame each time since the frame is information about the step currently running, and we don't want an outdated frame https://uwarg-docs.atlassian.net/wiki/spaces/CV/pages/2515959830/Logging+Module